packages/platforms/inventec/x86-64/x86-64-inventec-d7032q28b/onlp/builds/src/module/src/fani.c

```diff
@@ -0,0 +1,402 @@
+    MAKE_FAN_PATH_ON_PSU(10-0058)
+};
+
+#define MAKE_FAN_INFO_NODE_ON_MAIN_BOARD(id) \
+    { \
+        { ONLP_FAN_ID_CREATE(FAN_##id##_ON_MAIN_BOARD), "Chassis Fan "#id, 0 }, \
+        0x0, \
+        (ONLP_FAN_CAPS_SET_PERCENTAGE | ONLP_FAN_CAPS_GET_RPM | ONLP_FAN_CAPS_GET_PERCENTAGE), \
+        0, \
+        0, \
+        ONLP_FAN_MODE_INVALID, \
+    }
+
+#define MAKE_FAN_INFO_NODE_ON_PSU(psu_id, fan_id) \
+    { \
+        { ONLP_FAN_ID_CREATE(FAN_##fan_id##_ON_PSU##psu_id), "Chassis PSU-"#psu_id " Fan "#fan_id, 0 }, \
+        0x0, \
+        (ONLP_FAN_CAPS_SET_PERCENTAGE | ONLP_FAN_CAPS_GET_RPM | ONLP_FAN_CAPS_GET_PERCENTAGE), \
+        0, \
+        0, \
+        ONLP_FAN_MODE_INVALID, \
+    }
+
+/* Static fan information */
+onlp_fan_info_t linfo[] = {
+    { }, /* Not used */
+    MAKE_FAN_INFO_NODE_ON_MAIN_BOARD(1),
+    MAKE_FAN_INFO_NODE_ON_MAIN_BOARD(2),
+    MAKE_FAN_INFO_NODE_ON_MAIN_BOARD(3),
+    MAKE_FAN_INFO_NODE_ON_MAIN_BOARD(4),
+    MAKE_FAN_INFO_NODE_ON_MAIN_BOARD(5),
+    MAKE_FAN_INFO_NODE_ON_MAIN_BOARD(6),
+    MAKE_FAN_INFO_NODE_ON_PSU(1,1),
+    MAKE_FAN_INFO_NODE_ON_PSU(2,1),
+};
+
+#define VALIDATE(_id)                           \
+    do {                                        \
+        if(!ONLP_OID_IS_FAN(_id)) {             \
+            return ONLP_STATUS_E_INVALID;       \
+        }                                       \
+    } while(0)
+
+#define OPEN_READ_FILE(fd,fullpath,data,nbytes,len) \
+    DEBUG_PRINT("[Debug][%s][%d][openfile: %s]\n", __FUNCTION__, __LINE__, fullpath); \
+    if ((fd = open(fullpath, O_RDONLY)) == -1)  \
+       return ONLP_STATUS_E_INTERNAL;           \
+    if ((len = read(fd, r_data, nbytes)) <= 0){ \
+        close(fd);                              \
+        return ONLP_STATUS_E_INTERNAL;}         \
+    DEBUG_PRINT("[Debug][%s][%d][read data: %s]\n", __FUNCTION__, __LINE__, r_data); \
+    if (close(fd) == -1)                        \
+        return ONLP_STATUS_E_INTERNAL
+
+static uint32_t
+_onlp_fani_info_get_psu_fan_direction(void)
+{
+    /* Try to read direction from PSU1.
+     * If PSU1 is not valid, read from PSU2
+     */
+    int i = 0;
+
+    for (i = PSU1_ID; i <= PSU2_ID; i++) {
+        psu_type_t psu_type;
+        psu_type = get_psu_type(i, NULL, 0);
+
+        if (psu_type == PSU_TYPE_UNKNOWN) {
+            continue;
+        }
+
+		switch (psu_type) {
+			case PSU_TYPE_AC_F2B:
+			case PSU_TYPE_DC_48V_F2B:
+			case PSU_TYPE_DC_12V_F2B:
+				return ONLP_FAN_STATUS_F2B;
+			case PSU_TYPE_AC_B2F:
+			case PSU_TYPE_DC_48V_B2F:
+			case PSU_TYPE_DC_12V_B2F:
+				return ONLP_FAN_STATUS_B2F;
+			default:
+				return 0;
+		};
+    }
+
+    return 0;
+}
+
+static int
+_onlp_fani_info_get_fan(int local_id, onlp_fan_info_t* info)
+{
+    int   fd, len, nbytes = 10;
+    char  r_data[10]   = {0};
+    char  fullpath[65] = {0};
+
+    /* check if fan is present
+     */
+    sprintf(fullpath, "%s%s", PREFIX_PATH_ON_MAIN_BOARD, fan_path[local_id].present);
```

Comment:
Please remote the OPEN_READ_FILE() macros and replace with native calls to the onlp_file_* APIs only.